pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1227,34 +1235,40 @@ int parser_trim_buffers(parser_t *self) {
     size_t new_cap;
     void *newptr;
 
-    int i;
+    int64_t i;
 
     /* trim words, word_starts */
     new_cap = _next_pow2(self->words_len) + 1;
-    if ((int)new_cap < self->words_cap) {
+    if (new_cap < self->words_cap) {
         TRACE(("parser_trim_buffers: new_cap < self->words_cap\n"));
         newptr = safe_realloc((void *)self->words, new_cap * sizeof(char *));
         if (newptr == NULL) {
             return PARSER_OUT_OF_MEMORY;
         } else {
             self->words = (char **)newptr;
         }
-        newptr = safe_realloc((void *)self->word_starts, new_cap * sizeof(int));
+        newptr = safe_realloc((void *)self->word_starts,
+                              new_cap * sizeof(int64_t));
         if (newptr == NULL) {
             return PARSER_OUT_OF_MEMORY;
         } else {
-            self->word_starts = (int *)newptr;
+            self->word_starts = (int64_t *)newptr;
             self->words_cap = new_cap;
         }
     }
 
     /* trim stream */
-    new_cap = _next_pow2(self->stream_len) + 1;
+    if (new_cap < INT32_MAX) {
```

Comment:
Sorry, you found the last remaining holdout of early changes I made while determining what was causing the "oom" issue. This code is especially non-sensical since both branches do the same thing, but it's just cruft that needs to be removed at this point (and is in the latest commit).